src/wallet/rpcwallet.cpp

```diff
@@ -3740,11 +3766,11 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
 
     CScript scriptPubKey = GetScriptForDestination(dest);
     ret.pushKV("scriptPubKey", HexStr(scriptPubKey.begin(), scriptPubKey.end()));
-    const SigningProvider* provider = pwallet->GetSigningProvider();
+    const SigningProvider* provider = pwallet->GetSigningProvider(scriptPubKey);
 
     isminetype mine = pwallet->IsMine(dest);
     ret.pushKV("ismine", bool(mine & ISMINE_SPENDABLE));
-    bool solvable = IsSolvable(*provider, scriptPubKey);
+    bool solvable = provider ? IsSolvable(*provider, scriptPubKey) : false;
```

Comment:
In commit "Refactor: Require scriptPubKey to get wallet SigningProvider" (f7510c144b2262e1ee114eaa740b889c24ec7abb)

Would be slightly simpler to replace ternary with `solvable = provider && IsSolvable(*provider)`

Reply:
Done